### PR TITLE
Add OrOn operator for SqlJoinConditionBuilder

### PR DESCRIPTION
**Pull request by Yaraslaut**
Closes https://github.com/LASTRADA-Software/Lightweight/issues/132

### CODE REVIEW

**Comment by FelixTheC** [comment]
@Yaraslaut I had a similar Idea but @christianparpart thought about creating a parent class for the Where and Join Statements as they start sharing the same behavior. Like `And`, `Or` maybe we should also include `Not`.

File: src/tests/QueryBuilderTests.cpp [hunk 1]
```diff
@@ -216,6 +216,29 @@ TEST_CASE_METHOD(SqlTestFixture, "SqlQueryBuilder.FieldsWithBelongsTo", "[SqlQue
...
+            using namespace std::string_view_literals;
+            return q.FromTable("BETON_SORTE")
+                .Select()
+                .LeftOuterJoin("BEN_DESIGN"sv, "SORTE_NR"sv, "NR"sv)
+                .RightOuterJoin("BETON_ZEMENTART"sv,
+                                [](SqlJoinConditionBuilder q) {
+                                    return q.On("NR", { .tableName = "BETON_SORTE", .columnName = "ZEMENTART_NR" })
+                                        .Or("NR", { .tableName = "BETON_SORTE", .columnName = "ZEM_2_NR" })
+                                        .Or("NR", { .tableName = "BETON_SORTE", .columnName = "ZEM_3_NR" })
+                                        .Or("NR", { .tableName = "BETON_SORTE", .columnName = "ZEM_4_NR" });
```
Comment (christianparpart) [inline src/tests/QueryBuilderTests.cpp:232]:
```suggestion
                                        .OrOn("NR", { .tableName = "BETON_SORTE", .columnName = "ZEM_2_NR" })
                                        .OrOn("NR", { .tableName = "BETON_SORTE", .columnName = "ZEM_3_NR" })
                                        .OrOn("NR", { .tableName = "BETON_SORTE", .columnName = "ZEM_4_NR" });
```

Analogous to `Where` vs `OrWhere` - what do you two guys think about `On` vs `OrOn`?

Sounds more homogenous to me than `On` vs `Or`.

**Comment by Yaraslaut** [comment]
Ah, I see, it just this is a small changes, some repetition exist in query builder but it can be adapted later when api is more stable

File: src/tests/QueryBuilderTests.cpp (hunk 1 above)
Comment (Yaraslaut) [reply src/tests/QueryBuilderTests.cpp:232]: will change

**Review by christianparpart — APPROVED** [review]
I've rewritten the test case to not contain any business schema but rather generic table / column names.

**Pull request merged**